messenger/settings.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+import tarfile
+
+try: # Python 3 Compat
+    from urllib.request import urlretrieve
+    string = str
+except ImportError:
+    from urllib import urlretrieve
+    import string
+
+__all__= ['get_matlab_env', 'fetch_zmq']
+
+
+def get_matlab_env(matlab='matlab'):
+    """
+    Get the underlying environment variables set for a matlab installation.
+
+    Parameters
+    ----------
+    matlab: str
+        Path to the matlab binary executable.
+        If matlab is in the users $PATH, just pass 'matlab'
+
+    Returns
+    -------
+    environment: dict
+        Mapping of environment variable
+    """
+    command = ' '.join([matlab, '-e']),
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+
+    envs    = (line.decode('utf-8').strip() for line in process.stdout)
+
+    return dict(string.split(env, '=', maxsplit=1) for env in envs)
+
+
+def fetch_zmq(prefix, version=(4,0,5)):
```

Comment:
This might be useful in the long run.

I started writing code to do this, but was immediately overwhelmed with the gravity of building such an application.

It would require code for ensuring that proper build utilities are available on all platforms and then compiling zmq against that platform and then linking. Fortunately, this was actually successfully implemented by both pyzmq and iPython, so I do have their code as guidence.